### ISSUE TITLE
clone breaks gdb on power9

### ISSUE THREAD

**Issue by jmellorcrummey** [HPCToolkit/hpctoolkit#218]
@mwkrentel @blue42u 

surprise:
the shift from fork to clone causes problems with gdb on power9. (it may be a problem elsewhere as well, but we didn't check)

reproducer: 
```

hpcrun -e REALTIME@2000 -d /bin/ls &
gdb --pid $!
b main
call hpcrun_continue()
c
c
where
```

You will find that it doesn't know about any of the threads in the process or the prrocess call stack.

**Comment by blue42u** [comment]
I can get the main thread on x86 at least (on a hello world):
```
(gdb) call (void)hpcrun_continue()
(gdb) c
Continuing.
[New Thread 0x7f38c5ff94c0 (LWP 787853)]
process 784804 is executing new program: /home/deepthought/dyntest-suite/reference/hpctoolkit/install/libexec/hpctoolkit/hpcfnbounds-bin
[New LWP 784804]

Thread 3 "a.out" received signal SIGTRAP, Trace/breakpoint trap.
0x00007f38c6be1b41 in ?? () at dl-debug.c:74 from /lib64/ld-linux-x86-64.so.2
74	dl-debug.c: No such file or directory.
(gdb) b main
Breakpoint 1 at 0x11f9d0: file ../../../../../src/ref-hpctoolkit/src/tool/hpcfnbounds/main.cpp, line 143.
(gdb) c
Continuing.
Warning:
Cannot insert breakpoint 1.
Cannot access memory at address 0x11f9d0

Command aborted.
(gdb) i th
  Id   Target Id          Frame 
* 2    LWP 787853         0x00007f4fe26d929e in ?? ()
  3    LWP 784804 "a.out" 0x00007f38c6be1b41 in ?? () at dl-debug.c:74 from /lib64/ld-linux-x86-64.so.2
```
With this (hopefully similar) issue I know what's going on. Our two options are:
1. Add `CLONE_UNTRACED` to the `clone` flags, to prevent gdb from looking at our things. (Downside: gdb  will never be able to look at fnbounds)
2. Add `SIGCHLD` to the `clone` flags, to make it look to ptracers like a `fork`. (Downside: if fnbounds ever dies we have a rouge signal... like before *cough*)

Both work fine on my rig, just pick your poison.

**Comment by mwkrentel** [comment]
> 2. Add SIGCHLD to the clone flags, to make it look to ptracers like a
> fork. (Downside: if fnbounds ever dies we have a rouge signal...

What's wrong with SIGCHLD?  I wouldn't call it a rogue signal.  This
is the standard way fork() works.  The default action for SIGCHLD is
ignore, and we've been running this way with fork() for years.

**Comment by jmellorcrummey** [comment]
fixed  with commit 3df9041647c3b62d9414bf62917546a243845bdd
